telemetry/attributes.go

```diff
@@ -20,7 +22,7 @@ func attributeValueValid(val interface{}) bool {
 
 // vetAttributes returns the attributes that are valid.  vetAttributes does not
 // modify remove any elements from its parameter.
-func vetAttributes(attributes map[string]interface{}, errorLogger func(map[string]interface{})) map[string]interface{} {
+func vetAttributes(attributes map[string]interface{}) (map[string]interface{}, error) {
```

Comment:
This change does prevent us from needing to pass the logger around, and seems to make the api cleaner.